sktime/transformations/panel/_catch22_numba.py

```diff
@@ -487,26 +487,38 @@ def _spline_fit(X):
 
 
 @njit(fastmath=True, cache=True)
-def _normalise_series(X, mean):
+def _normalise_series(X: np.ndarray, mean):
     std = np.std(X)
     if std > 0:
         return (X - mean) / std
     return X
 
 
-def _DN_HistogramMode_5(X, smin, smax):
+def _perform_fft(series, smean):
+    nfft = int(np.power(2, np.ceil(np.log(len(series)) / np.log(2))))
+    return np.fft.fft(series - smean, n=nfft)
+
+
+@njit(fastmath=True, cache=True)
+def _DN_HistogramMode_5(series, smin, smax, smean, std, outlier_series, ac, acfz):
```

Comment:
can you explain why you give all features now eight arguments, `series`, `smin`, etc? Is this for performance reasons?

It is less crisp design-wise, as seven of these depend on the first.

If for performance reasons, have you tested performance?

Reply:
I wanted to omit passing args based on feature_id as in original `_transform_case` method, as it seemed overcomplicated and hard to debug. I tested using `numba.typed.Dict` but it was failing for multiple reasons. In theory only series and outlied_series could be passed, and the rest could be calculated later, but e.g., `_autocorellation` cannot be run in numba non-python mode, so then a wrapper for a method using this parameter should be created etc. So I thought it's better to pass all arguments for the time being, and improve it in the future. It's still better than:
```
if feature == 0 or feature == 1 or feature == 11:
                    if smin is None:
                        smin = np.min(series)
                    if smax is None:
                        smax = np.max(series)
                    args = [series, smin, smax]
                elif feature == 2 or feature == 22:
                    if smean is None:
                        smean = np.mean(series)
                    args = [series, smean]
                elif feature == 3 or feature == 4:
                    if self.outlier_norm:
                        if smean is None:
                            smean = np.mean(series)
                        if outlier_series is None:
                            outlier_series = _normalise_series(series, smean)
                        args = [outlier_series]
                    else:
                        args = [series]
                elif feature == 7 or feature == 8:
                    if smean is None:
                        smean = np.mean(series)
                    if fft is None:
                        nfft = int(
                            np.power(2, np.ceil(np.log(len(series)) / np.log(2)))
                        )
                        fft = np.fft.fft(series - smean, n=nfft)
                    args = [series, fft]
                elif feature == 5 or feature == 6 or feature == 12:
                    if smean is None:
                        smean = np.mean(series)
                    if fft is None:
                        nfft = int(
                            np.power(2, np.ceil(np.log(len(series)) / np.log(2)))
                        )
                        fft = np.fft.fft(series - smean, n=nfft)
                    if ac is None:
                        ac = _autocorr(series, fft)
                    args = [ac]
                elif feature == 16 or feature == 17 or feature == 20:
                    if smean is None:
                        smean = np.mean(series)
                    if fft is None:
                        nfft = int(
                            np.power(2, np.ceil(np.log(len(series)) / np.log(2)))
                        )
                        fft = np.fft.fft(series - smean, n=nfft)
                    if ac is None:
                        ac = _autocorr(series, fft)
                    if acfz is None:
                        acfz = _ac_first_zero(ac)
                    args = [series, acfz]
```
in my opinion

Reply:
I see. Yes, your first try was my inuition (see `Catch22Wapper`), but I see it interacts badly with `numba`. Fine with me for now, then.

Perhaps adding a short few lines of comments at the top of the block, about function signature, would be helpful to general developers looking at the file.

Reply:
> It's still better than:

Yes, of course.

Reply:
> I see. Yes, your first try aligns with my design intuition (see `Catch22Wapper`), but I see it interacts badly with `numba`. Fine with me for now, then.
> 
> Perhaps adding a short few lines of comments at the top of the block, about function signature, would be helpful to general developers looking at the file.

Sure, I'll add it! Passing it though a dict would be better I think, but e.g., every type of dict (in numba) must be the same, so e.g., a float has to be saved as single-element list of floats if other elements are floats. I'd have to delve into it